src/lib/datatypes/include/sol-list.h

```diff
@@ -66,9 +66,15 @@ struct sol_list {
 
 /**
  * @def SOL_LIST_INIT
- * @brief Helper macro to initialize a @c sol_list structure.
+ * @brief Helper macro to initialize a @c sol_list structure. Example:
+ *
+ * @code
+ * struct sol_list mylist = SOL_LIST_INIT(mylist);
+ * @endcode
+ *
+ * @param name List to be initialized
  */
-#define SOL_LIST_INIT { NULL, NULL }
+#define SOL_LIST_INIT(name) { &(name), &(name) }
 
```

Comment:
This is partially right. This is a kind of inlist, it means that this struct should be used in other struct to provide lists. There is no utiliity in use  it directly, as you can see the struct has two fields `prev` and `next` that with this macro will point to the same address. The usage of this is:

```
struct delayed_packet {
    struct sol_list list;
    struct sol_flow_packet *packet;
    uint16_t source_idx;
    uint16_t source_port_idx;
};
```

It said, i guess it is better initialize with the value given by the user and not get the address.